### DISCUSSION TITLE
Making "pre-commit" eslint faster with caching

### DISCUSSION THREAD

**Discussion by ikonst** [eslint/eslint#17973] [General]
In our orgs, developers have been complaining about the eslint part of the pre-commit checks taking "too long", which is still under 1 minute but noticeable.

Since we invariably all use IDEs that run eslint in the background, there's high probability the checks could be skipped with caching at pre-commit time. Unfortunately there's no way to apply caching uniformly through the config (we've decided not to do https://github.com/eslint/eslint/issues/13938). IntelliJ [seems to ignore](https://youtrack.jetbrains.com/issue/WEB-58978) the `--cache` arg somehow even if the dev is diligent to add it, but ideally I'd like it to require no configuration by individual devs.

Any suggestions?

**Comment by nzakas** [comment]
Sorry, there's really not enough information to know how to answer this. 

If you're running ESLint on precommit, then theoretically you're running it on a small number of files. If that's the case and it's still taking too long, then it's likely the plugins you're using that are the problem. For instance, `typescript-eslint` itself will slow things down dramatically, as will `eslint-plugin-import`. You may want to consider running a smaller subset of your config on precommit or else commenting out some things to see what you can do to speed it up.

**Reply by ikonst** [reply]
It's clearly typescript-eslint with type information that's slowing it down, that's known to us.

Running pre-commit with a subset of tests is a possible solution, but it comes with the cost of maintaining an alternative eslintrc (and being aware of that config quirk). What I was hoping is to avoid bifurcating the rule list, instead leveraging the fact that we almost invariably run eslint in the background by merit of editing those files in an IDE such as VS Code. If this eslint invocation would in turn update the (developer's local) cache, then when it's time to commit, this cache would likely short-circuit any need to 'check' the files.

**Reply by nzakas** [reply]
You're welcome to [submit an RFC](https://github.com/eslint/rfcs) if you have a design in mind. We just don't have the cycles to investigate and develop every suggestion that comes along.
